fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/NodePropertiesTools.java

```diff
@@ -194,37 +197,57 @@ public void removeReferencePlaceholders(final IdentifierConverter<Resource,Fedor
      */
     public void removeNodeProperty(final Node node, final String propertyName, final Value valueToRemove)
         throws RepositoryException {
-
+        LOGGER.debug("Request to remove {}", valueToRemove);
         // if the property doesn't exist, we don't need to worry about it.
         if (node.hasProperty(propertyName)) {
 
             final Property property = node.getProperty(propertyName);
+            final String strValueToRemove = valueToRemove.getString();
 
             if (property.isMultiple()) {
                 final AtomicBoolean remove = new AtomicBoolean();
-                final Value[] newValues = stream(node.getProperty(propertyName).getValues()).filter(v -> {
-                    if (v.equals(valueToRemove)) {
+                final Value[] newValues = stream(node.getProperty(propertyName).getValues()).filter(uncheck(v -> {
+                    final String strVal = v.getString();
+
+                    LOGGER.debug("v is '{}', valueToRemove is '{}'", v, strValueToRemove );
+                    if (strVal.equals(strValueToRemove)) {
                         remove.set(true);
                         return false;
                     }
+
+                    LOGGER.warn("Failed to find value for property '{}'. Unable to compare value (type: {}) " +
```

Comment:
Is this LOGGER statement correct? We are inspecting a multi-valued property here. If the value to be removed (`strValueToRemove`) does not equal `strVal`, all that means is that the value to remove is another one in the list.
For example, the multi-valued property may have the following values: `"a", "b", "c"`. If this request is trying to remove "b", then "a" and "c" will fall through to this LOGGER statement.
That is a perfectly expected case, not a "warning".